cumulus/node/src/service.rs

```diff
@@ -261,48 +275,48 @@ where
 
 	let code_executor = Arc::new(code_executor);
 
-	{
-		let spawn_essential = task_manager.spawn_essential_handle();
-		let (bundle_sender, bundle_receiver) = tracing_unbounded("transaction_bundle_stream");
-		let (execution_receipt_sender, execution_receipt_receiver) =
-			tracing_unbounded("execution_receipt_stream");
-
-		let executor = Executor::new(
-			primary_chain_client,
-			primary_network,
-			&spawn_essential,
-			select_chain,
-			imported_block_notification_stream,
-			new_slot_notification_stream,
-			client.clone(),
-			Box::new(task_manager.spawn_handle()),
-			transaction_pool,
-			Arc::new(bundle_sender),
-			Arc::new(execution_receipt_sender),
-			backend.clone(),
-			code_executor.clone(),
-			validator,
-			params.keystore_container.sync_keystore(),
-		)
-		.await?;
-
-		let executor_gossip =
-			cirrus_client_executor_gossip::start_gossip_worker(ExecutorGossipParams {
-				network: network.clone(),
-				executor,
-				bundle_receiver,
-				execution_receipt_receiver,
-			});
-		spawn_essential.spawn_essential_blocking("cirrus-gossip", None, Box::pin(executor_gossip));
-	}
-
-	Ok(NewFull {
+	let spawn_essential = task_manager.spawn_essential_handle();
+	let (bundle_sender, bundle_receiver) = tracing_unbounded("transaction_bundle_stream");
+	let (execution_receipt_sender, execution_receipt_receiver) =
+		tracing_unbounded("execution_receipt_stream");
+
+	let executor = Executor::new(
+		primary_chain_client,
+		primary_network,
+		&spawn_essential,
+		select_chain,
+		imported_block_notification_stream,
+		new_slot_notification_stream,
+		client.clone(),
+		Box::new(task_manager.spawn_handle()),
+		transaction_pool,
+		Arc::new(bundle_sender),
+		Arc::new(execution_receipt_sender),
+		backend.clone(),
+		code_executor.clone(),
+		validator,
+		params.keystore_container.sync_keystore(),
+	)
+	.await?;
+
+	let executor_gossip =
+		cirrus_client_executor_gossip::start_gossip_worker(ExecutorGossipParams {
+			network: network.clone(),
+			executor: executor.clone(),
+			bundle_receiver,
+			execution_receipt_receiver,
+		});
+	spawn_essential.spawn_essential_blocking("cirrus-gossip", None, Box::pin(executor_gossip));
+
+	let new_full = NewFull {
```

Comment:
Why not as part of `NewFull` like for everything else?

Reply:
I think the initial reason is not to make `NewFull` too complicated and deal with the compiling from the very beginning, but it can be solved much easier now, updated.